includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -993,14 +993,16 @@ private function validate_tag_spec_for_node( DOMElement $node, $tag_spec ) {
 
 		if ( ! empty( $tag_spec[ AMP_Rule_Spec::MANDATORY_PARENT ] ) && ! $this->has_parent( $node, $tag_spec[ AMP_Rule_Spec::MANDATORY_PARENT ] ) ) {
 			return [
-				'code' => self::WRONG_PARENT_TAG,
+				'code'                 => self::WRONG_PARENT_TAG,
+				'required_parent_name' => $tag_spec[ AMP_Rule_Spec::MANDATORY_PARENT ],
```

Comment:
Note that at some point we will not need to capture the `required_parent_name` in the error. We could instead capture the `spec_name` alone. Then the `spec_name` could be looked up to obtain the `required_parent_name` (or other properties depending on the error). This will be made possible by https://github.com/ampproject/amp-wp/pull/3817. However, that is not yet available, so what you are doing here is the perfect approach in the time being.

Reply:
Yep I initially was retrieving it by looking up the tag spec in `AMP_Allowed_Tags_Generated::class`, but I realized it was available in the sanitizer so I took it from there.

> This will be made possible by #3817.

Awesome, made a note of that.